src/Features/CSharp/Portable/ConvertToRecord/PositionalParameterInfo.cs

```diff
@@ -5,39 +5,78 @@
 using System.Collections.Immutable;
 using System.Linq;
 using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertToRecord
+namespace Microsoft.CodeAnalysis.CSharp.ConvertToRecord
 {
-    internal record PositionalParameterInfo(PropertyDeclarationSyntax Declaration, IPropertySymbol Symbol, bool KeepAsOverride)
+    /// <summary>
+    /// Represents a property that should be added as a positional parameter
+    /// </summary>
+    /// <param name="Declaration">Original declaration, null iff IsInherited is true
+    /// Null iff <see cref="IsInherited"/> is true</param>
+    /// <param name="Symbol">Symbol of the property</param>
+    /// <param name="KeepAsOverride">Whether we should keep the original declaration present</param>
+    /// <param name="IsInherited">Whether this property is inherited from another base record</param>
+    internal record PositionalParameterInfo(
+        PropertyDeclarationSyntax? Declaration,
+        IPropertySymbol Symbol,
+        bool KeepAsOverride,
+        bool IsInherited)
     {
         public static ImmutableArray<PositionalParameterInfo> GetPropertiesForPositionalParameters(
             ImmutableArray<PropertyDeclarationSyntax> properties,
             INamedTypeSymbol type,
             SemanticModel semanticModel,
             CancellationToken cancellationToken)
         {
-            // first get all property symbols
+            // get all declared property symbols, put inherited property symbols first
             var symbols = properties
                 .SelectAsArray(p => (IPropertySymbol)semanticModel.GetRequiredDeclaredSymbol(p, cancellationToken));
 
+            // add inherited properties from a potential base record first
+            var inheritedProperties = GetInheritedPositionalParams(type, cancellationToken);
+            var results = inheritedProperties.SelectAsArray(property => new PositionalParameterInfo(
+                Declaration: null,
+                property,
+                KeepAsOverride: false,
+                IsInherited: true));
+
             // The user may not know about init or be converting code from before init was introduced.
             // In this case we can convert set properties to init ones
             var allowSetToInitConversion = !symbols
                 .Any(symbol => symbol.SetMethod is IMethodSymbol { IsInitOnly: true });
 
-            return properties.ZipAsArray(symbols, (syntax, symbol)
+            results = results.Concat(properties.ZipAsArray(symbols, (syntax, symbol)
```

Comment:
Should this just be using a builder rather than repeatedly concatenating and allocating new arrays?